### PR TITLE
added spec change

### PR DESCRIPTION
**Pull request by ryandillinfelton**
## Overview

This PR is being created to address [this Jira](https://issues.redhat.com/browse/RHCLOUD-14534).
This is making our validation of MAC addresses stricter. The corosponding change in the inventory schemas repo has already been merged ([see here](https://github.com/RedHatInsights/inventory-schemas/pull/69)).

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by tahmidefaz — APPROVED** [review]
LGTM!

**Review by tahmidefaz — CHANGES_REQUESTED** [review]
Woops! That was too early. This schema doesn't match what's in `inventory-schemas`.

**Comment by ryandillinfelton** [comment]
> Woops! That was too early. This schema doesn't match what's in `inventory-schemas`.

Whoops, I must've made this branch earlier than I though. I've updated the regex to the version we just committed in schemas. Thanks for the catch :100:

File: swagger/system_profile.spec.yaml [hunk 1]
```diff
@@ -114,7 +114,7 @@ $defs:
         description: MAC address (with or without colons)
         example: "00:00:00:00:00:00"
         type: string
-        maxLength: 59
```
Comment (tahmidefaz) [inline swagger/system_profile.spec.yaml:117]: You don't need to remove this line.

**Review by tahmidefaz — CHANGES_REQUESTED** [review]
This still isn't the same as what we merged in `inventory-schemas`

File: swagger/system_profile.spec.yaml (hunk 1 above)
Comment (ryandillinfelton) [reply swagger/system_profile.spec.yaml:117]: Added it back in

File: swagger/system_profile.spec.yaml [hunk 2]
```diff
@@ -114,6 +114,7 @@ $defs:
         description: MAC address (with or without colons)
         example: "00:00:00:00:00:00"
         type: string
+        pattern: "^([A-Fa-f0-9]{2}[:-]){5}[A-Fa-f0-9]{2}$|^([A-Fa-f0-9]{4}[.]){2}[A-Fa-f0-9]{4}$|^[A-Fa-f0-9]{12}$|^([A-Fa-f0-9]{2}[:-]){19}[A-Fa-f0-9]{2}$|^[A-Fa-f0-9]{40}$"
```
Comment (kruai) [inline swagger/system_profile.spec.yaml:117]: Just wanted to note that every other `pattern` field in this document uses single quotes instead of double. Could we follow their example to keep consistency?

**Review by kruai — CHANGES_REQUESTED** [review]
One minor request

File: swagger/system_profile.spec.yaml (hunk 2 above)
Comment (ryandillinfelton) [reply swagger/system_profile.spec.yaml:117]: Sure

Comment (kruai) [reply swagger/system_profile.spec.yaml]: Ah, you're right; I was exclusively looking at the patterns. Looks good either way

**Review by kruai — APPROVED** [review]
LGTM

**Review by tahmidefaz — APPROVED** [review]
It's good now :+1:
